### ISSUE TITLE
Add best practices for dummy atoms

### ISSUE THREAD

**Issue by fjclark** [alchemistry/alchemical-best-practices#144]
# Proposed Change

Add a brief section in "What simulation protocol should I choose" on best practices for dummy atoms as detailed [here](https://pubs.acs.org/doi/full/10.1021/acs.jctc.0c01328). 

# Proposed Authors and Reviewers

Please confirm (before the end of January) that you are happy to author/ review and state when you're available. Authors, please say roughly when you plan to contribute the changes (no later than the end of May).

**Proposed authors**: @wiederm and/ or @sboresch
**Proposed reviewer**: Volunteer wanted

**Comment by sboresch** [comment]
Sure, I'll coordinate with  @wiederm!

**Comment by sboresch** [comment]
My suggestion of what to write/add here is part of pull request #158.

**Comment by fjclark** [comment]
I should have closed this when https://github.com/alchemistry/alchemical-best-practices/pull/158 was merged (as stated in the commit message). However, @yorkdm has now confirmed that he can provide some additional feedback, so I will keep this open a little while to allow him a bit longer to do so. I hope that's fine.

**Comment by davidlmobley** [comment]
@fjclark sorry, way behind on GitHub issues/notifications, but got here from your e-mail. I've just gone through that section again and accepted a number of changes and marked resolved some comments that were resolved. I think there are one or two changes from Michael Shirts that I need @sboresch to check as I'm not sure (I am not an expert on flapping). 

Did we decide whether/if we are adding section numbers to allow for easier cross-referencing of sections? cc @mrshirts

**Comment by fjclark** [comment]
@davidlmobley great, thank you, and no problem.

> Did we decide whether/if we are adding section numbers to allow for easier cross-referencing of sections? cc @mrshirts 

Yes, fully agree we need section numbers which were explicitly asked for by @sboresch and would allow some of @mrshirts' comments to be addressed. Sorry, has been on my to-do list but have been slow to get to it. A couple of options:

1. Keep the layout of section 6 as-is and add a new \subsubsubsection command to add numbering where we want it. I've added this purely for the Dummy atom section for now, which becomes 6.1.1.1. But, if we need subsubsub headings, I think we may want to restructure.

2. Restructure section 6 slightly to flatten out the structure and avoid subsubsub headings. For example, we currently have:

    6.1 Absolute and relative free energy calculations have important differences
        6.1.1 Choices unique to relative free energy calculations
        6.1.2 Absolute free energy calculations must handle the standard state and use binding site restraints

    Which could become e.g.

    6.1 Choices unique to relative free energy calculations
        6.1.1 Topologies
        ....
    6.2 Absolute free energy calculations must handle the standard state and use binding site restraints

    Current Section 6.2: "Absolute and relative calculations deal with some of the same issues" could also be flattened for consistency.


Also, section 6 is very large and I wonder if it should be split into two major sections -- the first covering 6.1: Absolute and relative free energy calculations have important differences, and the second covering 6.2: Absolute and relative calculations deal with some of the same issues (general simulations settings, multiple chemical states, etc). Alternatively, instead of framing in terms of differences/ commonalities between RBFE and ABFE, we could group by a) choices mainly related to picking end-states and b) choices mainly related to sampling/ transforming between end-states. Thought this would require a bit more work and would be better discussed on https://github.com/alchemistry/alchemical-best-practices/issues/172.

Please let me know your thoughts.

**Comment by sboresch** [comment]
@davidlmobley, I quickly checked your comments in the dummy atom section. All of this looks fine/OK to me (but see also direct mail!)

@fjclark , thanks for adding explicit numbering to the sections etc.!

Best,
Stefan

**Comment by fjclark** [comment]
Thanks for the input both. A few places it would be great to have more input are:

@sboresch:

1. At line 1056: "Somewhat unfortunately, Axelsen and Li, as well as Ries et al refer to their approach as dual topology. Ries et al.~suggested to use the term hybrid topology for (single coordinate) dual topology in the sense of Pearlman. We will attempt a disambiguation of the various nomenclatures below." David has commented: 

    "Should this be a footnote? This seems like a side note complaining about terminology ambiguity.

    (Personally, I decided the distinction between single topology and dual topology (and hybrid topology), if any, was confusing  enough that I just stopped using the terms and just try to explain what I'm DOING.)"

    Are you happy for this to be a footnote?

2. On line 1100: You've noted "\sbnote{I think this should get its own paragraph/subsection; to be expanded!}" Are you happy with it as-is? The software section might be better suited for a more detailed discussion of this.
3. Also on line 1100: "for example, AMBER TI uses a dual topology approach, while BioSimSpace uses a single topology approach." you've commented "Could the respective experts clarify whether dual topology is indeed a dual coordinate dual topology approach, and single topology a single coordinate single topology approach?". Lester Hedges (BioSimSpace) has confirmed that this is correct, while Ben Ries has said that AMBER TI "sounds like DCDT" from a reference (I have not been able to get Darrin York to comment). Shall we just drop the comment on AMBER given the lack of certainty, or are you happy to keep it?
4. Bullet points at line 1149 and 1150. There has been some rearrangement here which seems to have resulted in some repetition (" should test this as follows. Thus, a \underline{necessary} condition is as follows:") and Michael Shirts has moved some text to the next bullet point. Please could you check you're happy with these bullet points?
5. Bullet point at line 1152: "While solvation free energies can nowadays be computed quite fast...". This was deleted by David yesterday (and temporarily reintroduced by me now for discussion), though there was some confusion in the comments. You've written: "Here I am confused: is one \item already removed? I guess stating that cycle closure is a tedious, but stringent test to check for dummy atom related errors would be OK. And referencing the older Loeffler et al paper is actually a good idea!" -- could you please update this bullet based on your comment?

@davidlmobley:

1. At line 1142: "E.g., when mutating methane to ammonia, one of the methane hydrogen atoms becomes a dummy atom. According to the three-degrees-of-freedom rule, it should be connected to the nitrogen only by one distance and two bond angle terms. However, these three internal coordinates permit the dummy atom to adopt several positions in Cartesian coordinates. In simulations one will find the dummy atom not only at the position of the former hydrogen atom, but also other positions. Indeed, it could even have a position overlapping with one of three ammonia hydrogens." Stefan has commented "Not sure that this example clarifies anything. Alternatively, we could point to Figs. 6 & 7 of the Fleck et al. dummy atom paper." Do you have any comments?


@davidlmobley  and @sboresch:

1. Discussion of the use of "arbitrarily" on line 1142.  How should this be worded? Would this cover it?
> In the latter case, the factorization of the partition function is not exact, but the overall error can still be made arbitrarily small, at least for the systems tested in Ref.~\cite{Fleck_2021}. 
2. I've made the dummy atom section and others "subsubsubsections" so they're numbered. However, as mentioned above, I wonder if we should flatten out the structure slightly (https://github.com/alchemistry/alchemical-best-practices/issues/144#issuecomment-4050560643). What do you think.
3. Comments on the atom mapping points discussed here https://github.com/alchemistry/alchemical-best-practices/issues/145#issuecomment-4027134821 would also be helpful.


Thanks very much!

**Comment by sboresch** [comment]
@fjclark , thanks for your comments; I'll try to address as much as possible quickly below:

1) _At line 1056: "Somewhat unfortunately, ..._
_..._
_Are you happy for this to be a footnote?_

First, I made a minor change in the text. It may be a quibble about terminology, but it is/was one of the factors making me introduce the 'spiel' about single vs dual _**coordinate**_ approaches in the first place. So, I would prefer this to remain in the main text, otherwise the final sentence of this paragraph "We will attempt a disambiguation of the various nomenclatures below." makes little sense.

2) _On line 1100: You've noted "\sbnote ..._

I had hoped that people familiar with the main programs, GROMACS and AMBER, etc., would add some lines here, but since this did not happen or happened in the software section, I think this is fine as is. I guess unless you get Darrin to confirm / chime in, we should drop the AMBER line, though I then wonder whether the whole paragraph(?) should be dropped ...

3) Your items 4 & 5, the 'best practices bullet list': Indeed, something got out of sync here; I have rewritten this a bit so that it makes a bit more sense. Note that this is far from perfect, and not everything is 'a best practice recommendation:'

I wonder whether the following commented line

%Further work automating the correct setup and treatment of dummy atoms is necessary.

should not be readded, possibly quoting 'Ghost'?

4) Concerning 'arbitrarily', l 1142: I like your suggestion! By qualifying the 'arbitrarily small' only for the "toy systems" we analyzed hopefully puts this into perspective.

5) For the numbering of (sub)subsections, I would be pragmatic, i.e., proceed as follows: For the time being, leave numbers in, at least for sections, subsections and subsubsections. Once this is _really_ quite final, check whether any of the subsubsections are even explicitly referenced. If not, then we can drop these numbers. I don't know on top of my head how to do this in Latex, but I have certainly done this in the past.

6) I don't feel competent to add to the atom mapping section.

**Comment by fjclark** [comment]
Thanks for the quick response @sboresch.

1. Thanks, unless @davidlmobley objects (comment on footnote at line 1065), I'll mark this as resolved.
2. Thanks, I'll try Darrin again.
3. Great, thank you. If I get time soon and nobody else has added it, I'll try to suggest something pointing to ghostly and the unsettled practices section.
5. I've modified this as suggested.
6. Thanks, I'll avoid changing the structure for the moment. Yes, it's easy to change the level where numbering stops. Currently, there are quite a few cross-references to e.g. the dummy atoms subsubsubsection.

**Comment by fjclark** [comment]
## Updates

@sboresch, on point 2: @yorkdm has updated this this to

> for example, AMBER~\cite{Case_JChemInfModel_2025_v65_p7835} formally uses a DCDT approach but can constrain the positions of selected "common core" atoms to be the same (effectively mimicking the behavior of a SCST for these atoms)

and I've marked the comment as resolved.

On point 3: I've pointed to ghostly and the unsettled practices section:

> Further work automating the correct setup and treatment of dummy atoms is necessary (see also Sec. \ref{sec:unsettled}); the package ghostly (Sec. \ref{sec:software}) is a step towards this \cite{hedges_ghostly_2025.2.0}.

## Issues to Address

@sboresch, could you please address the points below:

a) @yorkdm has commented 

> FYI: I noticed that "DCDT" was used several times (dual coordinate dual topology) - but earlier at the top of the section "Summary, comparison of single and dual coordinate methods, orthogonal classifications:", it is listed as just "DC".  I didn't change that to "DCDT" because I wasn't sure what was intended - but unless I missed something, DCDT is never defined...

b) You still have an uncommented note at line 1102: "\sbnote{Check whether this is correct!}.", referring to:

> Some studies refer to a ``hybrid'' topology approach to free energy calculations~\cite{gapsys2015pmx, gapsys2016accurate, gapsys2020large}, though in our view these calculations should be classified as single coordinate single topology

@vgapsys are you happy with the above? Alternatively, could you please check this @sboresch?

c) Please feel free to reword line 1174 ("Further work automating the correct setup..."), and please feel free to mark the changes as accepted once you're happy.

d) On line 1158, you've commented on the example "E.g., when mutating methane to ammonia":
> Not sure that this example clarifies anything. Alternatively, we could point to Figs. 6 & 7 of the Fleck et al. dummy atom paper.

  Could you please update this as you see fit as a tracked change and I'll ask Michael to review soon?

e) Comment chain around line 1171 starting with David's "I don't understand what this is referring to or why this is in the list.". Sorry about the confusion due to this no longer pointing to anything in the article. As far as I can tell this is now resolved. Please mark it as resolved if so, or I can ping David via email if required.

f) While we're here, there are a couple of unaddressed comments from Michael on lines 1032 and 1042. Please could you respond, or since these are minor points, mark them as resolved if you strongly disagree?

Thanks very much.

There are a few other contentious additions around this section concerning TIES, but these are probably best dealt with separately.

**Comment by sboresch** [comment]
> ## Updates
[ snip]
> 
> ## Issues to Address
> 
> [@sboresch](https://github.com/sboresch), could you please address the points below:
> 
> a) [@yorkdm](https://github.com/yorkdm) has commented
> 
> > FYI: I noticed that "DCDT" was used several times (dual coordinate dual topology) - but earlier at the top of the section "Summary, comparison of single and dual coordinate methods, orthogonal classifications:", it is listed as just "DC".  I didn't change that to "DCDT" because I wasn't sure what was intended - but unless I missed something, DCDT is never defined...

I am not finding the comment in the manuscript, but indeed 'dual coordinate dual topology' was used rarely in the manuscript, as in this case 'dual topology' seems redundant. I quickly scanned the manuscript and replaced all residual uses of DCDT or 'dual coordinate dual topology' by DC and 'dual coordinate', respectively. In some places, this left "... in the dual coordinate method" which doesn't sound right as there is not just one dual coordinate method, so I changed this to 'dual coordinate method/approach'.

> b) You still have an uncommented note at line 1102: "\sbnote{Check whether this is correct!}.", referring to:
> 
> > Some studies refer to a ``hybrid'' topology approach to free energy calculations~\cite{gapsys2015pmx, gapsys2016accurate, gapsys2020large}, though in our view these calculations should be classified as single coordinate single topology
> 
> [@vgapsys](https://github.com/vgapsys) are you happy with the above? Alternatively, could you please check this [@sboresch](https://github.com/sboresch)?

I am fine with this -- this comment was mostly intended to give others the option to clarify / correct me. I commented out the \sbnote{}.
> 
> c) Please feel free to reword line 1174 ("Further work automating the correct setup..."), and please feel free to mark the changes as accepted once you're happy.
> 

Accepted
> d) On line 1158, you've commented on the example "E.g., when mutating methane to ammonia":
> 
> > Not sure that this example clarifies anything. Alternatively, we could point to Figs. 6 & 7 of the Fleck et al. dummy atom paper.
> 
> Could you please update this as you see fit as a tracked change and I'll ask Michael to review soon?
> 

I resolved my comment, after making two minor changes to my text (visible in tracking mode). Indeed, someone other than me should review this. Note that is now l.1157!

> e) Comment chain around line 1171 starting with David's "I don't understand what this is referring to or why this is in the list.". Sorry about the confusion due to this no longer pointing to anything in the article. As far as I can tell this is now resolved. Please mark it as resolved if so, or I can ping David via email if required.
> 

I took the liberty to resolve this comment. The 'bullet list' starting currently on l.1163 of the Overleaf version of the manuscript looks OK to me. Since the comment thread between David and myself was  hard to follow and in part outdated, it's thus removed.

> f) While we're here, there are a couple of unaddressed comments from Michael on lines 1032 and 1042. Please could you respond, or since these are minor points, mark them as resolved if you strongly disagree?
> 

l.1032, now 1031: I concur that this paragraph is mostly redundant. I believe I didn't touch it as this is from V1 of the manuscript.

l.1042, now 1041: I responded to the comment, with which I somewhat disagree. If I had written the whole section as a 'historical review', then this would indeed be inappropriate. But it seemed worthwhile to stress that the meaning of 'dual topology' changed significantly over the years, motivating my substantial rewrite.  

> Thanks very much.
> 
> There are a few other contentious additions around this section concerning TIES, but these are probably best dealt with separately.

**Comment by fjclark** [comment]
Thanks very much @sboresch.

> l.1032, now 1031: I concur that this paragraph is mostly redundant. I believe I didn't touch it as this is from V1 of the manuscript.

I've removed this as you and Michael agree.

I've marked minor changes as accepted throughout. While everything else (barring TIES additions) appears resolved, I've left larger comments with your responses to Michael unresolved and will close this issue once he takes a last look over.

**Comment by vgapsys** [comment]
@sboresch @fjclark 
I like the topology classification in Section 6.2.1.1 and Figure 5, yet, in its current form it is incomplete. In particular, the classification would not cover many contemporary approaches, e.g. FEP+, openFE, pmx, biosimspace. These approaches cannot be classified as SCST, because they would not follow the leftmost perturbation path in Figure 5, rather, for the benzene - benzylalcohol perturbation they would look like SCDT approach. However, the SCDT approach has a requirement that "_no atoms are allowed to change type_". This would, of course, be violated by many common algorithms, as they allow perturbations of the heavy atom types (or, in some cases, even heavy atom to hydrogen perturbations).

To my understanding, we need an intermediate category of _Single Coordinate Hybrid Topology_: it would cover most of the contemporary methods, which have a set of rules encoded in their atom mapper to decide which atom perturbations are allowed. For example, if we change the Figure 5 example to pyridine -> benzylalcohol, a mapper could make a C->N atom pair between the ligands: this way the atom type will be perturbed directly and the topology looks more like SCST. However, C->H perturbation is usually not allowed and goes via introduction of dummies, this way we get SCDT approach. Since such different mappings appear in one pair of ligands the topology could be classified as _hybrid_, rather than strictly _single_ or _double_.

I added a paragraph explaining this on the line 1183. If this looks ok, we should also update the lines 1212-1223

**Comment by sboresch** [comment]
Thanks, Vytautas!

Your paragraph looks good. I will also remove the \emph{} on no atoms are allowed to change type since this may have been the intent, but never was very practical.

Please double-check my edits.

**Comment by fjclark** [comment]
Thanks both!

@vgapsys could you please check @sboresch's edits? If you're happy I'll close the issue (we'll deal with TIES additions separately).

**Comment by vgapsys** [comment]
@sboresch  @fjclark 
sorry for the delayed reply, the edits look good to me!

**Comment by fjclark** [comment]
Thanks all, closing this as complete! (Issues related to TIES are being dealt with separately)
